src/utils/getFilenameFromUrl.js

```diff
@@ -95,8 +95,14 @@ function getFilenameFromUrl(context, url) {
         filename = path.join(outputPath, querystring.unescape(pathname));
       }
 
-      let fsStats;
+      if (
+        options.historyApiFallback &&
+        !context.outputFileSystem.existsSync(filename)
+      ) {
+        filename = path.join(outputPath);
+      }
```

Comment:
When pathname is empty, with this option we will also try find index.html at correct folder

Reply:
@alexander-akait  This PR.